### ISSUE TITLE
POIs are not reliable

### ISSUE THREAD

**Issue by TROMsite** [FacilMap/facilmap#189]
I keep on seeing the message "Zoom in to show POIs" and only rarely it works to show them. See this as an example https://maps.trom.tf/zG11ITedVruZnqij#15/42.4340/1.9214/HiBi-o_drinkingwater_hospital_pharmacy_police_toilets_alpinehut_campsite_hostel_diyshop_generalshop_shoeshop_mall_textileshop_bakery_groceryshop_supermarket_bench_clinic_picnic_chalet

**Comment by cdauth** [comment]
Right now the POIs are loaded from an external server, and there is a hard limit in place in order not to overload that server (and potentially get banned). Loading so many types of POIs at once exceeds those limits by far.

I think right now the feature is only really usable to display one or two types of POIs at once.

Maybe in the future I will introduce a button to temporarily raise the limit once. But then when you move around the map, new POIs will not be loaded automatically, because that will definitely overload the server if too many people are using it.

**Comment by TROMsite** [comment]
Ah I see....are they loaded from your server or a 3rd party service?

**Comment by cdauth** [comment]
They are loaded from this service at the moment: https://overpass.kumi.systems/

**Comment by TROMsite** [comment]
I see. Is there no way to load the POI from our server? Could be a good idea to let the facilmap instances use their own overpass instance if possible. Like I had to create APIs for the other services that are required by facilmap in order to work properly. Else it will be limited.

**Comment by cdauth** [comment]
In theory it would be nice if FacilMap was hosting everything itself, like the search, the routing, the tiles and the Overpass API. Then I could finally implement a proper search and a proper cycling map. But I think the resources required for this are immense. For the Overpass API alone it is recommended to have 32 GB of RAM and 300 GB of disk space.

Considering how few users (I think) FacilMap has at the moment and that there is zero income, I feel like it is neither worth the effort nor the money to do this.

**Comment by TROMsite** [comment]
Oh wow 32 GB of RAM and 300 GB of disk space....that's insane. For now it is what it is I guess.

**Comment by cdauth** [comment]
Yes, but not surprising considering how big the world is and how detailed the OSM database is by now.

**Comment by TROMsite** [comment]
32 GB of RAM and 300 GB means it can do its own search and routes? But for how many users? That's some 20 Euros a month, not as expensive if it can support a lot of users.

**Comment by cdauth** [comment]
That's just for the Overpass API. So only for the POIs. It might be able to run the search as well, but I'm not sure how well that would work.

For Nominatim, which is currently used for the search, [64 GB of RAM and 900 GB of disk space](https://nominatim.org/release-docs/latest/admin/Installation/#hardware) are required.

For OSRM, which is currently used for routing without details, something like [248 GB of RAM and 350 GB of disk space](https://github.com/Project-OSRM/osrm-backend/wiki/Disk-and-Memory-Requirements) are required.

To host one set of tiles, [48 GB of RAM and 1 TB of disk space](https://help.openstreetmap.org/questions/64405/tile-server-hardware-requirements/64406) are needed.

It seems unrealistic to even host multiple of these on the same server. If FacilMap had hundreds of thousands of users and a stable income of donations, I would consider spending the time and money to do that, but at the moment that is something really unachievable.

**Comment by TROMsite** [comment]
Paints a better picture now. Thanks fore explaining. You need a few powerful servers and for now, you are right, is unachievable.
